repose-aggregator/components/services/datastore-service/datastore-service-api/src/main/resources/META-INF/schema/config/hazelcast-datastore.xsd

```diff
@@ -0,0 +1,222 @@
+  You may obtain a copy of the License at
+  
+       http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+  -->
+
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           xmlns:html="http://www.w3.org/1999/xhtml"
+           xmlns="http://docs.openrepose.org/repose/hazelcast-datastore/v1.0"
+           targetNamespace="http://docs.openrepose.org/repose/hazelcast-datastore/v1.0"
+           attributeFormDefault="unqualified"
+           elementFormDefault="qualified">
+
+    <xs:element name="hazelcast-datastore" type="HazelcastDatastoreConfig"/>
+
+    <xs:complexType name="HazelcastDatastoreConfig">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    Hazelcast Datastore settings.
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:choice>
+            <xs:element name="standard" type="StandardConfig"/>
+            <xs:element name="simplified" type="SimplifiedConfig"/>
+        </xs:choice>
+    </xs:complexType>
+
+    <xs:complexType name="StandardConfig">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    A hook that allows for usage of a standard Hazelcast configuration.
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:attribute name="href" type="xs:anyURI" use="required">
+            <xs:annotation>
+                <xs:documentation>
+                    <html:p>
+                        A direct URI reference to a standard Hazelcast XML configuration.
+                    </html:p>
+                </xs:documentation>
+            </xs:annotation>
+        </xs:attribute>
+    </xs:complexType>
+
+    <xs:complexType name="SimplifiedConfig">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    A custom subset of the full Hazelcast configuration meant to simplify usage.
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:all>
+            <xs:element name="port" type="Port" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="join" type="Join" minOccurs="1" maxOccurs="1"/>
+        </xs:all>
+    </xs:complexType>
+
+    <xs:complexType name="Port">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    The ports which Hazelcast will use to communicate between cluster members. Its default value is
+                    5701.
+                </html:p>
+                <html:p>
```

Comment:
We'd normally put these notes on the attribute itself. What do these actually do? Can it not run off a single port?

Reply:
It can run off a single port. I believe these attributes just enable Hazelcast to bind to more than one port. I know one case where it will do so is if the selected port is unavailable. I am not sure what cases would cause Hazelcast to bind to multiple ports.

See:
https://docs.hazelcast.org/docs/3.12/manual/html-single/index.html#port

Reply:
Oh, as for the notes, that was just the result of copy-pasting the type definition. I will move them.

Reply:
yeah, i think single port will do.